File: Champions/Lucian/Q.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onStartCasting()
	local current = Vector2:new(owner.X, owner.Y)
    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    local range = to * 1100
    local trueCoords = current + range
	
	addLaser(trueCoords.x, trueCoords.y, true)
	spellAnimation("SPELL1", owner)
	addParticle(owner, "Lucian_Q_laser.troy", trueCoords.x, trueCoords.y)
	addParticleTarget(owner, "Lucian_Q_tar.troy", owner)

end
function onFinishCasting()

end

function applyEffects()
	local damage = (owner:GetStats().AttackDamage.Total*(0.45 + (spellLevel*0.15))) + (50 + (spellLevel*30))
	dealPhysicalDamage(damage)
	
end

function onUpdate(diff)

end

File: Champions/Lucian/Q.cs
using System.Numerics;
using GameServerCore.Enums;
using LeagueSandbox.GameServer.API;
using GameServerCore.Domain.GameObjects;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
using LeagueSandbox.GameServer.GameObjects.Missiles;
using GameServerCore.Domain;
using LeagueSandbox.GameServer.Scripting.CSharp;

namespace Spells
{
    public class LucianQ : IGameScript
    {
        public void OnActivate(IChampion owner)
        {
        }

        public void OnDeactivate(IChampion owner)
        {
        }

        public void OnStartCasting(IChampion owner, ISpell spell, IAttackableUnit target)
        {
            var current = new Vector2(owner.X, owner.Y);
            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            var range = to * 1100;
            var trueCoords = current + range;

            spell.AddLaser("LucianQ", trueCoords.X, trueCoords.Y);
            spell.SpellAnimation("SPELL1", owner);
            ApiFunctionManager.AddParticle(owner, "Lucian_Q_laser.troy", trueCoords.X, trueCoords.Y);
            ApiFunctionManager.AddParticleTarget(owner, "Lucian_Q_cas.troy", owner);
        }

        public void OnFinishCasting(IChampion owner, ISpell spell, IAttackableUnit target)
        {
        }

        public void ApplyEffects(IChampion owner, IAttackableUnit target, ISpell spell, IProjectile projectile)
        {
            var damage = owner.Stats.AttackDamage.Total * (0.45f + spell.Level * 0.15f) + (50 + spell.Level * 30);
            target.TakeDamage(owner, damage, DamageType.DAMAGE_TYPE_PHYSICAL, DamageSource.DAMAGE_SOURCE_SPELL,
                false);
        }

        public void OnUpdate(double diff)
        {
        }
    }
}
